src/Profile.cpp

```diff
@@ -412,11 +412,15 @@ namespace geopm
         geopm_time(&overhead_entry);
 #endif
 
-        if (hint && ((hint & (hint - 1)) != 0)) {   /// power of 2 check
-            throw Exception("ProfileImp:region() multiple region hints set and only 1 at a time is supported.",
-                            GEOPM_ERROR_RUNTIME, __FILE__, __LINE__);
-        }
+        geopm::check_hint(hint);
         uint64_t result = m_table->key(region_name);
+
+#ifdef GEOPM_DEBUG
+        if (m_region_ids.find(result) == m_region_ids.end()) {
```

Comment:
Use emplace(), rather than the find/insert combo.

https://en.cppreference.com/w/cpp/container/unordered_set/emplace